packages/animations/lib/src/open_container.dart

```diff
@@ -0,0 +1,636 @@
+          break;
+        case AnimationStatus.forward:
+        case AnimationStatus.reverse:
+          break;
+      }
+    });
+
+    return super.didPush();
+  }
+
+  @override
+  bool didPop(void result) {
+    _takeMeasurements(
+      navigatorContext: subtreeContext,
+      delayForSourceRoute: true,
+    );
+    return super.didPop(result);
+  }
+
+  void _takeMeasurements({
+    BuildContext navigatorContext,
+    bool delayForSourceRoute = false,
+  }) {
+    final RenderBox navigator =
+        Navigator.of(navigatorContext).context.findRenderObject();
+    final Size navSize = _getSize(navigator);
+    _sizeTween.end = navSize;
+
+    void takeMeasurementsInSourceRoute([Duration _]) {
+      final Rect srcRect = _getRect(hideableKey, navigator);
+      _sizeTween.begin = srcRect.size;
+      _insetsTween.begin = Rect.fromLTRB(
+        srcRect.left,
+        srcRect.top,
+        navSize.width - srcRect.right,
+        navSize.height - srcRect.bottom,
+      );
+      hideableKey.currentState.placeholderSize = _sizeTween.begin;
+    }
+
+    if (delayForSourceRoute) {
+      SchedulerBinding.instance
+          .addPostFrameCallback(takeMeasurementsInSourceRoute);
+    } else {
+      takeMeasurementsInSourceRoute();
+    }
+  }
+
+  Size _getSize(RenderBox render) {
+    assert(render != null && render.hasSize);
+    return render.size;
+  }
+
+  Rect _getRect(GlobalKey key, RenderBox ancestor) {
+    final RenderBox render = key.currentContext.findRenderObject();
+    assert(render != null && render.hasSize);
+    assert(ancestor != null && ancestor.hasSize);
+    return MatrixUtils.transformRect(
+      render.getTransformTo(ancestor),
+      Offset.zero & render.size,
+    );
+  }
+
+  bool get _transitionWasInterrupted {
+    bool wasInProgress = false;
+    bool isInProgress = false;
+
+    switch (_currentAnimationStatus) {
+      case AnimationStatus.completed:
+      case AnimationStatus.dismissed:
+        isInProgress = false;
+        break;
+      case AnimationStatus.forward:
+      case AnimationStatus.reverse:
+        isInProgress = true;
+        break;
+    }
+    switch (_lastAnimationStatus) {
+      case AnimationStatus.completed:
+      case AnimationStatus.dismissed:
+        wasInProgress = false;
+        break;
+      case AnimationStatus.forward:
+      case AnimationStatus.reverse:
+        wasInProgress = true;
+        break;
+    }
+    return wasInProgress && isInProgress;
+  }
+
+  void closeContainer() {
+    Navigator.of(subtreeContext).pop();
+  }
+
+  @override
+  Widget buildPage(
+    BuildContext context,
+    Animation<double> animation,
+    Animation<double> secondaryAnimation,
+  ) {
+    return Align(
+      alignment: Alignment.topLeft,
+      child: AnimatedBuilder(
+        animation: animation,
+        builder: (BuildContext context, Widget child) {
+          if (animation.isCompleted) {
+            return SizedBox.expand(
+              child: Material(
```

Comment:
What if the OpenBuilder returns a widget with a shape? Wouldn't this Material need to have the same shape?

Reply:
Added an openShape parameter. Good idea!